cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -350,6 +351,22 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 		serviceaccountcontroller.DefaultServiceAccountsControllerOptions(),
 	).Run()
 
+	if s.HealthCheckControllerSyncPeriod > 0 {
+		config := client.AddUserAgent(kubeconfig, "healthchecker-controller")
+
+		checkerTransport, err := client.TransportFor(config)
+		if err != nil {
+			panic(err)
+		}
+		httpClient := &http.Client{Transport: checkerTransport}
+		healthChecker := healthzcheckcontroller.NewHTTPSHealthzChecker(httpClient, "healthz")
+
+		healthCheckClient := clientset.NewForConfigOrDie(config)
+		checkController := healthzcheckcontroller.NewAPIServerHealthCheckController(
+			healthCheckClient.Core().Endpoints("default"), healthChecker, "kubernetes", recorder)
```

Comment:
The api server takes a parameter for what namespace has the master kubernetes service.

Do we need to add a new arg to match kube-apiserver:
```
fs.StringVar(&s.MasterServiceNamespace, "master-service-namespace", s.MasterServiceNamespace, "The namespace from which the kubernetes master services should be injected into pods")
```